spec/services/e_invoices/payments/factur_x/create_service_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe EInvoices::Payments::FacturX::CreateService do
+  let(:organization) { create(:organization, premium_integrations: %w[issue_receipts]) }
+  let(:payment) { create(:payment, organization:) }
+  let(:xml_builder_double) { instance_double(Nokogiri::XML::Builder, to_xml: xml_content) }
+  let(:xml_content) { "<xml>content</xml>" }
+
+  around { |test| lago_premium!(&test) }
+
+  describe "#call" do
+    context "when payment exists" do
+      it "builds the XML" do
+        allow(Nokogiri::XML::Builder).to receive(:new).with(encoding: "UTF-8")
```

Comment:
Do we need to mock here ? I believe we can call the build and simply assess some of the expected values.